redis-server.py

```diff
@@ -1,23 +1,150 @@
 # redis-server.py
 
 import socket
+import argparse
+import threading
+import time
 
 from utils import server_response_decode
 
 HOST = "127.0.0.1"  # Standard loopback interface address (localhost)
-PORT = 65430  # Port to listen on (non-privileged ports are > 1023)
-MY_DICT = {}
+PORT = 6380  # Port to listen on (non-privileged ports are > 1023)
+DATA_DICT = {}  # Dictionary for storage of data on "Redis" server
+TIME_DICT = {}  # Dictionary for storing key and time (value) for SET time based options
+BUFFER_SIZE = 1024
+SEPARATOR = "\r\n"
+BULK_STR_START = "$"
+NIL_REPLY = "$-1\r\n"
+SYNTAX_ERROR_MSG = "-ERR syntax error\r\n"
+TIME_OPTIONS = set(["EX", "PX", "EXAT", "PXAT", "KEEPTTL"])
+SET_OPTIONS = set(["EX", "PX", "EXAT", "PXAT", "KEEPTTL", "XX", "NX", "GET"])
+OKAY = "+OK\r\n"
 
-with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
-    s.bind((HOST, PORT))
-    s.listen()
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--port",
+        "-p",
+        type=int,
+        default=PORT,
+        help="This is the default port to connect to",
+    )
+    return parser.parse_args()
+
+
+def listen_and_respond():
+    args = parse_args()
+    with socket.socket() as s:
+        s.bind((HOST, args.port))
+        s.listen()
+        # Creates infinite while statement that accepts incoming connections and transmits data between
+        while True:
+            conn, addr = s.accept()
+            # print(f"Connected by {addr}")
+            data = conn.recv(BUFFER_SIZE).decode()
+            # if no data is sent the statements are broken and script ends.
+            if not data:
+                break
+            decoded_response, _ = server_response_decode(data)
+            # parse command and either retrieve/store data or return error message
+            RESP_encoded_response = command_handler(decoded_response, DATA_DICT)
+            conn.sendall(f"{RESP_encoded_response}".encode())
+    return data
+
+
+def unknown_command_error_response(message):
+    return f"-ERR unknown command '{' '.join(message)}'{SEPARATOR}"
+
+
+def bulk_string_response(length, response):
+    return f"{BULK_STR_START}{length}{SEPARATOR}{response}{SEPARATOR}"
+
+
+def command_handler(message, data_dictionary):
+    # generates bulk string, simple string, null reply and integer reply.
+    if type(message) != list:
+        return unknown_command_error_response(message)
+    if message[0] == "GET":
+        if len(message) > 2:
```

Comment:
`!= 2` would be a better check because this currently allows a call like `cli> GET`